src/app/features/courses/courses.component.html

```diff
@@ -1,23 +1,31 @@
-<ng-container *ngIf="courseInfo; else courseList">
-  <!-- Render course info -->
-  <app-course-info (showCourseList)="handleShowCourses()" [course]="courseInfo"></app-course-info>
+<ng-container *ngIf="isLoading$ | async; then Loading; else coursesTemplate">
+
 </ng-container>
 
-<ng-template #courseList>
+<ng-template #coursesTemplate>
   <ng-container *ngIf="courses.length>0; else emptyList" >
-    <app-search (search)="handleSearch($event)" [placeholder]="'Course...'"></app-search> 
+    <div class="actionBar">
+      <app-search (search)="handleSearch($event)" [placeholder]="'Course...'"></app-search>
+      <ng-container *ngIf="isAdmin$ | async">
+        <app-button (click)="onAddButtonClick()" [type]="ButtonTypes.Button" [buttonText]="'ADD NEW COURSE'"></app-button>
+      </ng-container>
+    </div>
+
     <ng-container *ngIf="notFound">
       <span [ngClass]="'noCourse'">No course found for {{lastSearchedText}} - All courses are shown</span>
     </ng-container>
-    <app-courses-list (showCourse)="handleShowCourseInfo($event)" [editable]="true" [courses]="filteredCourses"></app-courses-list>
+    <app-courses-list [courses]="filteredCourses"></app-courses-list>
   </ng-container>
 </ng-template>
 
 <ng-template #emptyList>
   <app-info [title]="emptyListTitle" [text]="emptyListText">
-    <app-button [type]="ButtonTypes.Button" [buttonText]="'ADD NEW COURSE'"></app-button>
+    <app-button (click)="onAddButtonClick()" [type]="ButtonTypes.Button" [buttonText]="'ADD NEW COURSE'"></app-button>
   </app-info>
 </ng-template>
 
+<ng-template #Loading>
+  <span>LOADING</span>
```

Comment:
Once it is 'loading' I cannot move anywhere 😆 
It is happening upon course delete. I suppose the course shall be deleted and user should be navigated back to course list.
![Uploading image.png…]()

Reply:
I corrected this yesterday, dont know how I failed to save or what happened :).
Problem is just that if the deletion is successful in the observable next, the loading is not set to false